pkg/csi/service/wcp/controller_helper.go

```diff
@@ -115,65 +116,62 @@ func validateWCPControllerUnpublishVolumeRequest(ctx context.Context, req *csi.C
 // ExpandVolumeRequest for WCP CSI driver. Function returns error if validation
 // fails otherwise returns nil.
 func validateWCPControllerExpandVolumeRequest(ctx context.Context, req *csi.ControllerExpandVolumeRequest,
-	manager *common.Manager, isOnlineExpansionEnabled bool) error {
+	manager *common.Manager) error {
 	log := logger.GetLogger(ctx)
 	if err := common.ValidateControllerExpandVolumeRequest(ctx, req); err != nil {
 		return err
 	}
 
-	if !isOnlineExpansionEnabled {
-		var nodes []*vsphere.VirtualMachine
+	var nodes []*vsphere.VirtualMachine
```

Comment:
This entire code needs to be removed. It checks if any Pods in GC are using the WCP PVC and stops online expansion.

Reply:
IIUC, for online expansion, we want the expansion to go through even if pods in GC are using it. And, volume expansion has already been enabled for WCP. Then, why do we have this method present? Does this method currently block any online expansion in WCP?

Reply:
It was introduced when WCP supported offline expansion only. Now that you are removing dead code, you need to remove this entire function.

Reply:
But, we would still need the function `ValidateControllerExpandVolumeRequest()` as it is checking for volume ID and capacity ranges? https://github.com/adikul30/vsphere-csi-driver/blob/rm-fss-online-volume-extend/pkg/csi/service/wcp/controller_helper.go#L121-L123

Reply:
Oh yes, we can keep that. Sorry, my bad

Reply:
cool. It's the same for vanilla too. I think only gc has the appropriate check at the moment.